server.py

```diff
@@ -165,14 +165,25 @@ def license():
 if __name__ == '__main__':
 
     if os.path.isfile(certfile) and os.path.isfile(keyfile):
+        ssl_ctx = ssl.create_default_context()
+        ssl_ctx.load_cert_chain(certfile, keyfile)
+
+        # Protocol options: allow TLSv1.1 and later
+        ssl_ctx.options |= ssl.OP_NO_SSLv2
+        ssl_ctx.options |= ssl.OP_NO_SSLv3
+        ssl_ctx.options |= ssl.OP_NO_TLSv1
+
+        # Cipher options: strong ciphers, follow server preferences
+        ssl_ctx.set_ciphers("ECDH+AESGCM128:ECDH+AESGCM256:ECDH+AES128:ECDH+AES256:!aNULL:!MD5:!RC4:!DSS:!EXPORT")
```

Comment:
What's the rationale?

- Specifying the signature algorithm is pointless, since we only have a RSA key.
Once we remove the superfluous stuff, Mozilla's list becomes
> ECDHE-AES256-GCM-SHA384:ECDHE-CHACHA20-POLY1305:ECDHE-AES128-GCM-SHA256:ECDHE-AES256-SHA384:ECDHE-AES128-SHA256

- The version of OpenSSL in the container doesn't support Chacha20:
> ECDHE-AES256-GCM-SHA384:ECDHE-AES128-GCM-SHA256:ECDHE-AES256-SHA384:ECDHE-AES128-SHA256

- The current cipher string expands to
> ECDHE-AES128-GCM-SHA256:ECDHE-AES256-GCM-SHA256:ECDHE-AES128-SHA256:ECDHE-AES128-SHA:ECDHE-AES256-SHA256:ECDHE-AES256-SHA

  We have two differences with the Mozilla list:
    - use of HMAC-SHA1 is still allowed: doing otherwise causes some compatibility issues, and it's not currently a problem, crypto-wise;
    - AES128 is preferred over AES256, for a good reason: there are crypto issues that only affect AES256 (and AES192).